### PR TITLE
Add emojis and several more symbols as word separators, fix #113404

### LINKED ISSUE #113404: Wrong highlight if there are duplicate strings in completion item
- VSCode Version: 1.52.1
- OS Version: MacOS Big Sur 11.1

If a suggestion contains two "div", the second "div" is wrongly highlighted after "di" is entered

div

Does this issue occur when all extensions are disabled?: No, with my customed completion extension

### PR DESCRIPTION
**Pull request by HaoboGu**
This PR adds emojis and several more symbols as word separators. VSCode highlights matches after added symbols and emojis correctly now.

This PR fixes #113404

### CODE REVIEW

File: src/vs/base/common/filters.ts [hunk 1]
```diff
@@ -468,21 +468,7 @@ function isSeparatorAtPos(value: string, index: number): boolean {
...
-		case CharCode.Backslash:
-		case CharCode.SingleQuote:
-		case CharCode.DoubleQuote:
-		case CharCode.Colon:
-		case CharCode.DollarSign:
-			return true;
-		default:
-			return false;
-	}
+	return isWordSeparator(code) || strings.isEmojiImprecise(code);
```
Comment (jrieken) [inline src/vs/base/common/filters.ts:471]: I see how it is inconsistent of us having two ways to identify word separators but I would not make this change (removal of switch-case) here. For once the sets of characters are different and then I believe that using a set over a switch-case is slower

**Review by jrieken — CHANGES_REQUESTED** [review]

File: src/vs/base/common/filters.ts (hunk 1 above)
Comment (HaoboGu) [reply src/vs/base/common/filters.ts:471]: Thanks for the comment! I'm still working on testing which symbol should be added as a separator. I saw some test cases failed and I haven't found the cause

**Review by jace4511 — APPROVED** [review]

**Review by jrieken — COMMENTED** [review]
lgtm

**Comment by jrieken** [comment]
Thanks

**Pull request merged**